packages/core/src/graph/graph.ts

```diff
@@ -1783,14 +1785,52 @@ export default abstract class AbstractGraph extends EventEmitter implements IAbs
       comboConfig = combo;
     }
 
-    // step2: 更新 children，根据类型添加 comboId 或 parentId
+    // step 2: Pull children out of their parents
```

Comment:
The `createCombo` function is now over 110 lines long. Do you think it could be in scope of this PR to split into functions (one for each step)? It's fine if not.

Reply:
I'd prefer not do it unless they ask - the way I'm reading the existing codebase long functions are not a problem.